scripts/create_x509_chain_crypto.py

```diff
@@ -0,0 +1,558 @@
+        inter_key=intermediate_private_key,
+        device_common_name="device" + common_name,
+        password=device_password,
+        key_size=key_size,
+        days=days,
+    )
+
+
+def create_root_ca_cert(root_common_name, root_private_key, days=3650):
+    file_root_certificate = "demoCA/newcerts/ca_cert.pem"
+
+    root_public_key = root_private_key.public_key()
+
+    subject = x509.Name(
+        [x509.NameAttribute(NameOID.COMMON_NAME, str.encode(root_common_name).decode("utf-8"))]
+    )
+
+    builder = create_cert_builder(
+        subject=subject, issuer_name=subject, public_key=root_public_key, days=days, is_ca=True
+    )
+
+    root_cert = builder.sign(
+        private_key=root_private_key, algorithm=hashes.SHA256(), backend=default_backend()
+    )
+    with open(file_root_certificate, "wb") as f:
+        f.write(root_cert.public_bytes(serialization.Encoding.PEM))
+
+    return root_cert
+
+
+def create_intermediate_ca_cert(
+    root_cert_subject, root_key, intermediate_common_name, intermediate_private_key, days=3650
+):
+    file_intermediate_certificate = "demoCA/newcerts/intermediate_cert.pem"
+    file_intermediate_csr = "demoCA/newcerts/intermediate_csr.pem"
+
+    intermediate_csr = create_csr(
+        private_key=intermediate_private_key,
+        csr_file=file_intermediate_csr,
+        subject=intermediate_common_name,
+        is_ca=True,
+    )
+
+    builder = create_cert_builder(
+        subject=intermediate_csr.subject,
+        issuer_name=root_cert_subject,
+        public_key=intermediate_csr.public_key(),
+        days=int(days / 10),
+        is_ca=True,
+    )
+
+    intermediate_cert = builder.sign(
+        private_key=root_key, algorithm=hashes.SHA256(), backend=default_backend()
+    )
+    with open(file_intermediate_certificate, "wb") as f:
+        f.write(intermediate_cert.public_bytes(serialization.Encoding.PEM))
+
+    return intermediate_cert
+
+
+def create_multiple_device_keys_and_certs(
+    number_of_devices,
+    inter_cert_subject,
+    inter_key,
+    device_common_name,
+    password,
+    key_size=4096,
+    days=3650,
+):
+
+    for i in range(1, number_of_devices + 1):
+        device_password_file = COMMON_DEVICE_PASSWORD_FILE + str(i) + EXTENSION_NAME
+        device_csr_file = COMMON_DEVICE_CSR_FILE + str(i) + EXTENSION_NAME
+        device_cert_file = COMMON_DEVICE_CERT_FILE + str(i) + EXTENSION_NAME
+        device_private_key = create_private_key(
+            password_file=device_password_file, password=password, key_size=key_size
+        )
+        device_csr = create_csr(
+            private_key=device_private_key,
+            csr_file=device_csr_file,
+            subject=device_common_name + str(i),
+            is_ca=False,
+        )
+
+        builder = create_cert_builder(
+            subject=device_csr.subject,
+            issuer_name=inter_cert_subject,
+            public_key=device_csr.public_key(),
+            days=int(days / 100),
```

Comment:
>days=int(days / 100) [](start = 12, length = 20)

this on is divided by 100.  this all seems very arbitrary and random.

Reply:
not actually random...also this exactly does what the last script used to do.....so this is how i go with it..roots are created for 1 year, intermediate for almost a  month, device for almost some days...

---
In reply to: [432503675](https://github.com/Azure/azure-iot-sdk-python/pull/574#discussion_r432503675) [](ancestors = 432503675)